Lib/tempfile.py

```diff
@@ -822,8 +822,17 @@ def resetperms(path):
 
                     try:
                         _os.unlink(path)
+                    except IsADirectoryError:
+                        cls._rmtree(path, ignore_errors=ignore_errors)
                     # PermissionError is raised on FreeBSD for directories
-                    except (IsADirectoryError, PermissionError):
+                    except PermissionError:
+                        # On Windows, calling _rmtree again will raise
+                        # IsADirectoryError and mask the PermissionError.
```

Comment:
I'm not sure, but it should be ok to raise `IsADirectoryError` for non-file stuff right? If we can't clear a folder due to both `PermissionError` and `IsADirectoryError`, I think it's okay to raise either exception. It's not too clear which one should take precedence.

In the bpo, @akulakov suggested only re-raising for files since `IsADirectoryError` is definitely wrong in those cases.

Reply:
The issue is about **NotA**DirectoryError, but your change and your comment only mention **IsA**DirectoryError. I'm still confused.

Reply:
OMG, I feel like an idiot now 😆 . Thanks for pointing that out. Yep the comment is wrong, it should be about `NotADirectoryError`.